internal/command/command.go

```diff
@@ -0,0 +1,221 @@
+// here we can do all boilerplate code that is else copied in each command and make sure
+// we have one place to handle all errors and ensure commands have consistent results
+func Add(c *cobra.Command, command Command) {
+	command.GetCmd().RunE = func(cmd *cobra.Command, args []string) error {
+		// initialize project but ignore error since config can be missing
+		project, _ := flow.LoadProject(flow.ConfigPath)
+
+		gateway, err := createGateway(cmd, project)
+		handleError("Gateway Error", err)
+
+		logger := createLogger()
+
+		service := services.NewServices(gateway, project, logger)
+
+		// run command
+		result, err := command.Run(cmd, args, project, service)
+		handleError("Command Error", err)
+
+		// format output result
+		formattedResult, err := formatResult(result)
+		handleError("Result", err)
+
+		// output result
+		err = outputResult(formattedResult)
+		handleError("Output Error", err)
+
+		return nil
+	}
+
+	bindFlags(command)
+	c.AddCommand(command.GetCmd())
+}
+
+// createGateway creates a gateway to be used, defaults to grpc but can support others
+func createGateway(cmd *cobra.Command, project *flow.Project) (gateway.Gateway, error) {
+	// create in memory emulator client
+	if RunEmulatorFlag {
+		return gateway.NewEmulatorGateway(), nil
+	}
+
+	// resolve host
+	host := HostFlag
+	if host == flow.DefaultHost && project != nil {
+		if NetworkFlag != "" {
+			check := project.GetNetworkByName(NetworkFlag)
+			if check == nil {
+				return nil, fmt.Errorf("provided network with name %s doesn't exists in condiguration", NetworkFlag)
+			}
+
+			host = project.Host(NetworkFlag)
+		} else {
+			host = project.Host("emulator")
+		}
+	} else if host == "" {
+		return nil, fmt.Errorf("host must be provided using --host flag or in config by initializing project: flow project init")
+	}
+
+	// create default grpc client
+	return gateway.NewGrpcGateway(host)
+}
+
+// create logger utility
+func createLogger() util.Logger {
+	// disable logging if we user want a specific format like JSON
+	//(more common they will not want also to have logs)
+	verbose := util.InfoLog
+	switch LogFlag {
+	case "none":
+		verbose = util.NoneLog
+	case "error":
+		verbose = util.ErrorLog
+	case "debug":
+		verbose = util.DebugLog
+	default:
+		verbose = util.InfoLog
+	}
+
+	if FormatFlag != "" {
+		verbose = util.NoneLog
+	}
+
+	return util.NewStdoutLogger(verbose)
+}
+
+// outputResult takes care of showing the result
+func formatResult(result Result) (string, error) {
+	if result == nil {
+		return "", fmt.Errorf("Missing")
+	}
+
+	if FilterFlag != "" {
+		var jsonResult map[string]interface{}
+		val, _ := json.Marshal(result.JSON())
+		err := json.Unmarshal(val, &jsonResult)
+		if err != nil {
+			return "", err
+		}
+
+		return fmt.Sprintf("%v", jsonResult[FilterFlag]), nil
+	}
+
+	switch FormatFlag {
+	case "json":
+		jsonRes, _ := json.Marshal(result.JSON())
+		return string(jsonRes), nil
```

Comment:
Could do something like:

`return filterOutput(FilterFlag, jsonRes)`

Reply:
Here the filter flag doesn't matter anymore as commented above